### ISSUE TITLE
Avatar fine tuning

### ISSUE THREAD

**Issue by cstiens** [zom/Zom-iOS#232]
@N-Pex there are a couple minor issues with the avatar image rotating and being squished.

1. When I choose 'Take a Photo or Video' and take a photo in portrait mode, it rotates 90 degrees to be horizontal in the avatar circle.
2. On that note, shouldn't this just say 'Take a photo' so that people don't think they can have video avatars? (Side note: Video avatars would be super cool.)
3. When I choose 'Photo Library', and add a portrait picture, it also rotates 90 degrees. However, when I select a portrait screenshot, it does not rotate it.
4. Every portrait image is squished. Landscape images seem to crop just fine.

It would be nice to have these things fixed before we release this version on the App Store. Can you advise on how complicated it would be, and if it makes sense to do it before merging OMEMEO?

**Comment by chrisballinger** [comment]
@davidchiles could probably fix this fastest because he wrote the image resizing code / tests

**Comment by cstiens** [comment]
@chrisballinger Thank you! @davidchiles will you have a look at this?

**Comment by cstiens** [comment]
@davidchiles I'm having the same issues in the latest build Version 1.0.13 (62). In addition, landscape images now seems to be 'squished' in the avatar view.

**Comment by N-Pex** [comment]
@davidchiles Just trying out things after the merge. I think that I am able to "set" the avatar image in onboarding, but the UI is never updated. Should we call "refreshAvatarImage" at "//We updated the avatar"?

**Comment by davidchiles** [comment]
@cstiens Unsure if the changes made it in the current testflight. @N-Pex What commit was the testflight built from? If it includes 05aefcd2d0b729752d67a670c1ee05ac2a6597ac then there is a bug.
@N-Pex I'll take a look at setting the avatar during onboarding.

**Comment by cstiens** [comment]
@davidchiles portrait photos (from Camera and Photo Library) are 'squishing' when they are cropped, but everything else seems to work well on my device on my first quick test of the avatars.

**Comment by davidchiles** [comment]
Waiting on https://github.com/ChatSecure/ChatSecure-iOS/pull/700 to pass Travis then it'll be merged and we can point Zom to the most recent chatsecure commit. The issue was images from the iPhone camera have [imageOrientation](https://developer.apple.com/reference/uikit/uiimage/1624141-imageorientation) properties our cropping rect needs to be rotated to match the orientation of the original image.

**Comment by N-Pex** [comment]
@davidchiles I am confused on which repo/branch to use from now on?

**Comment by chrisballinger** [comment]
@N-Pex Submit PRs to upstream/master, but make sure not to submit them from zom/master because it fouls up the back merge. Flow should look roughly like this:

1. Create feature branch of the "Zom/ChatSecure-iOS" fork like `expose-headers`. Zom's submodule now points to `zom/expose-headers`, instead of `zom/master`.
2. Submit PR from `zom/expose-headers` to `upstream/master`.
3. We merge PR upstream, then pull in `upstream/master` into `zom/master` via fast-forward.
4. Zom submodule now points at `zom/master` again.

**Comment by cstiens** [comment]
fixed. bug filed for avatar crashing the app on ipads.
